toolkit/tools/pkg/profile/profile.go

```diff
@@ -0,0 +1,81 @@
+package profile
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"runtime/pprof"
+	"runtime/trace"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/exe"
+)
+
+type Profiler struct {
+	cpuProfFile *os.File
+	memProfFile *os.File
+	traceFile   *os.File
+}
+
+func StartProfiling(pf *exe.ProfileFlags) (*Profiler, error) {
+	p := &Profiler{
+		cpuProfFile: nil,
+		memProfFile: nil,
+		traceFile:   nil,
+	}
+
+	if *pf.EnableCpuProf {
+		cpf, err := os.Create(*pf.CpuProfFile)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to create cpu-pprof file: %s", err)
```

Comment:
Linter will probably be grumpy about the capitalization of "Unable". It likes error strings to be all lowercase.